gen/src/write.rs

```diff
@@ -1575,30 +1594,52 @@ fn write_rust_vec_extern(out: &mut OutFile, key: NamedImplKey) {
 
 fn write_rust_option_extern(out: &mut OutFile, inner: OptionInner) {
     out.include.cstddef = true;
+    let value_type;
+    let mut is_const = false;
     let element = match inner {
```

Comment:
`is_const` being `mut` here feels a little unnecessary. It would read a bit clearer if every branch specified exactly if `is_const` is true or false. You could also consider assigning these three values as a tuple

```
let (element, value_type, is_const) = match inner { ... };
```